resources/config.rb

```diff
@@ -0,0 +1,115 @@
+#
+# Cookbook:: resolver
+# Resource:: config
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+include Resolver::Cookbook::Helpers
+
+property :config_file, String,
+          default: lazy { RESOLVER_RESOLV_CONF_DEFAULT },
+          description: 'The path to the resolver configuration file on disk'
+
+property :cookbook, String,
+          default: 'resolver',
+          description: 'Template source cookbook'
+
+property :template, String,
+          default: 'resolv.conf.erb',
+          description: 'Template source file'
+
+property :owner, String,
+          default: lazy { RESOLVER_CONF_USER_DEFAULT },
+          description: 'Generated file owner'
+
+property :group, String,
+          default: lazy { resolver_conf_group_default },
+          description: 'Generated file group'
+
+property :mode, String,
+          default: '0644',
+          description: 'Generated file mode'
+
+property :nameservers, [String, Array],
+          coerce: proc { |p| p.is_a?(String) ? p.split(',') : p },
+          required: true,
+          description: 'The DNS servers to configure for system name resolution'
+
+property :domain, String,
+          description: 'The DNS domain name for the host'
+
+property :search, [String, Array],
+          coerce: proc { |p| p.is_a?(String) ? p.split(',') : p },
+          description: 'The DNS domain search list for the host',
+          callbacks: {
+            'with glibc < 2.26 the number of search domains are limited to 6' => lambda { |p|
+              p.count <= 6 || (node['languages']['c']['glibc']['version'].to_f >= 2.26)
+            },
+            'with glibc < 2.26 the total length of the search list is limited to 256 character' => lambda { |p|
+              p.join(' ').length <= 256 || (node['languages']['c']['glibc']['version'].to_f >= 2.26)
+            },
+          }
+
+property :sortlist, [String, Array],
+          coerce: proc { |p| p.is_a?(Array) ? p : [p] },
+          description: 'The DNS sort list for the host'
+
+property :options, Hash,
+          description: 'Additional options to add to the resolver configuration file'
+
+property :override_system_configuration, [true, false],
```

Comment:
I originally made this a configurable setting in my previous pr to make the cookbook functional without being a breaking change, but I think this should always be true for this resource going forward.

Reply:
The default is set to true for 99.9% of cases and anyone can flip it back to `false` if they don't want the functionality so I think we're good there?

Reply:
It's just a thing that I think no one would ever touch in practice. If you're using this resource, you should want your config to be respected. It's fine though, not worth preventing a merge

Reply:
I get you. On the flip side I'm a big fan of always providing as many nerd knobs as you can as there'll be an edge case somewhere eventually that you'd never preconceive where it'll be useful. It's easier to modify a role than a cookbook code change.